tdrs-backend/tdpservice/users/serializers.py

```diff
@@ -57,31 +57,24 @@ class Meta:
 
 
 class UserProfileSerializer(serializers.ModelSerializer):
-    """Serializer used for setting a user's profile."""
+    """Serializer used for retrieving/updating a user's profile."""
 
-    stt = STTUpdateSerializer(required=True)
-    email = serializers.SerializerMethodField("get_email")
+    email = serializers.CharField(read_only=True, source='username')
     roles = GroupSerializer(
-        many=True, required=False, allow_empty=True, source="groups"
+        many=True,
+        read_only=True,
+        source='groups'
     )
+    stt = STTPrimaryKeyRelatedField()
```

Comment:
Using a nested serializer here that was not read-only was causing a 500 error on certain requests:
```
Exception Type: AssertionError at /v1/users/set_profile/
Exception Value: The `.update()` method does not support writable nested fi=
elds by default.
Write an explicit `.update()` method for serializer `tdpservice.users.seria=
lizers.UserProfileSerializer`, or set `read_only=3DTrue` on nested serializ=
er fields.
```

Furthermore, we don't want to allow users to update the actual STT record - just the STT associated with their record. Using this subclass of PrimaryKeyRelatedField allows us to ensure that users can set the STT via just the ID and automatically adds checks to ensure that the requested ID actually exists.